src/commonMain/kotlin/fr/acinq/lightning/payment/OfferManager.kt

```diff
@@ -183,7 +183,10 @@ class OfferManager(val nodeParams: NodeParams, val walletParams: WalletParams, v
                         logger.warning { "offerId:${offer.offerId} pathId:${decrypted.pathId} ignoring invoice request, could not build onion message: ${invoiceMessage.value}" }
                         sendInvoiceError("failed to build onion message", decrypted.content.replyPath)
                     }
-                    is Right -> OnionMessageAction.SendMessage(invoiceMessage.value)
+                    is Right -> {
+                        logger.info { "sending BOLT 12 invoice for ${invoice.amount} to introduction node ${destination.route.introductionNodeId}" }
```

Comment:
Can we also log the whole invoice? This way we have everything we need to troubleshoot potential issues.
It shouldn't be too spammy.

```suggestion
                        logger.info { "sending BOLT 12 invoice for ${invoice.amount} to introduction node ${destination.route.introductionNodeId} (invoice=${invoice.toString()})" }
```

Reply:
I thought about it but the invoice can be very big as we copy all fields from the invoice request. An attacker could make the invoice as large as the largest onion message. Do you still think we should log it?
I would be more comfortable adding specific fields.

Reply:
I hesitated exactly because of that...it's true that the `invoice_request` can be arbitrarily big because of the padding TLV. Let me check what other fields may be useful for debugging.

Reply:
You're right, there are a lot of thing that aren't very useful for debugging, let's just also include `paymentHash` and `payerId` and we should have the most useful fields!

Reply:
Maybe we can just truncate the invoice at a fixed length? That should be enough for debugging normal users

Reply:
That wouldn't be useful, if we truncate it we can't decode it...